### PR TITLE
Allow links in the disco add-on descriptions

### LINKED ISSUE #3213: Allows links in the summary for recommended add-ons.
Currently recommended add-ons have links removed by DOMPurify:

add-ons_manager

We need to look to allow links in the sanitization step so they don't get removed from the content entirely.

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #3213

### CODE REVIEW

File: tests/unit/disco/components/TestAddon.js [hunk 1]
```diff
@@ -245,6 +246,31 @@ describe(__filename, () => {
       );
     });
 
+    it('purifies an editorial description with a bad link', () => {
```
Comment (tofumatt) [inline tests/unit/disco/components/TestAddon.js]: Thanks for testing this! 👍

File: tests/unit/disco/components/TestAddon.js [hunk 2]
```diff
@@ -245,6 +246,31 @@ describe(__filename, () => {
...
+      const data = {
+        ...result,
+        description: 'This is a <a href="javascript:alert(1)">description</a>',
+      };
+      const root = renderAddon({ addon: data, ...data });
+      expect(root.find('.editorial-description').html()).toContain(
```
Comment (kumar303) [inline tests/unit/disco/components/TestAddon.js]: For future reference, @tofumatt discovered that you can do this which is pretty handy:

```js
const anchor = root.find('.editorial-description').render().find('a');
```

Comment (tofumatt) [reply tests/unit/disco/components/TestAddon.js]: Hah, shoot, that's true, I didn't even think of that. I always think to do it myself but forget when reviewing 😅

Would be nice to do that before merging.

Comment (muffinresearch) [reply tests/unit/disco/components/TestAddon.js]: What's the intermediate render for?  Couldn't you do `root.find('.editorial-description a')`

Comment (muffinresearch) [reply tests/unit/disco/components/TestAddon.js]: I'm going to land this for now but I'm not against doing a separate update to fix the entire file for consistency if we're going to make any changes to how these assertions are done once I understand the rationale a bit better. Presumably the idea would be to move to more object representation based assertions?

Comment (kumar303) [reply tests/unit/disco/components/TestAddon.js]: > Couldn't you do `root.find('.editorial-description a')`

It doesn't work, possibly in relation to this [bug](https://github.com/airbnb/enzyme/issues/419).

> I'm not against doing a separate update to fix the entire file for consistency

I personally don't think it's worth a follow-up. Your assertion works fine and will be easy enough to maintain. There are other situations where an HTML based assertion could be more of a maintenance burden.

**Pull request merged**